### PR TITLE
Migrate libboost 1.82

### PR DESCRIPTION
**Pull request by h-vetinari**
Sounds like the piggyback migrator hit a corner-case, do it manually.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -120,13 +120,13 @@ outputs:
...
         - zlib        # [build_platform != target_platform]
       host:
-        - boost-cpp {{ boost_cpp }}
         - cairo
         - curl
         - fontconfig
         - freetype
         - gettext
         - glib
+        - libboost-devel
```
Comment (h-vetinari) [inline recipe/meta.yaml:129]: use `libboost-devel` because of the run-dep (now handled through the run-export)...

File: recipe/meta.yaml [hunk 2]
```diff
@@ -188,13 +187,13 @@ outputs:
...
         - zlib        # [build_platform != target_platform]
       host:
-        - boost-cpp {{ boost_cpp }}
         - cairo
         - curl
         - fontconfig
         - freetype
         - gettext
         - glib
+        - libboost-headers
```
Comment (h-vetinari) [inline recipe/meta.yaml:196]: And `libboost-headers` where there's not run-dep (and hence no run-requirement). Due to the layered build here, we should still verify that the link-check agrees.

Comment (h-vetinari) [reply recipe/meta.yaml:196]: All builds have:
```
WARNING (poppler): dso library package conda-forge::libboost-1.82.0-hf0c313a_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```
so indeed, we don't need `-devel` here.

**Comment by h-vetinari** [comment]
@conda-forge/poppler, this should be ready, PTAL! :)

**Review by xhochy — APPROVED** [review]

**Pull request merged**